ArduPlane/mode_auto.cpp

```diff
@@ -11,6 +11,8 @@ bool ModeAuto::_enter()
     plane.next_WP_loc = plane.prev_WP_loc = plane.current_loc;
     // start or resume the mission, based on MIS_AUTORESET
     plane.mission.start_or_resume();
+    // initialize speed variable used in Auto for DO_CHANGE_SPEED commands
+    plane.new_airspeed_cm = -1;
```

Comment:
also set to -1 in the exit for AUTO and GUIDED

Reply:
add it to Mode::enter() to cover all modes

Reply:
done